File: linkedinsdk/src/main/java/com/AbdAllahAbdElFattah13/linkedinsdk/ui/linkedin_builder/LinkedInFromActivityBuilder.java
package com.AbdAllahAbdElFattah13.linkedinsdk.ui.linkedin_builder;

import android.app.Activity;
import android.content.Intent;

public final class LinkedInFromActivityBuilder extends LinkedInBuilder {

  private Activity activity;

  private LinkedInFromActivityBuilder(Activity activity) {
    super(activity);
    this.activity = activity;
  }

  public static LinkedInFromActivityBuilder getInstance(Activity activity) {
    return new LinkedInFromActivityBuilder(activity);
  }

  @Override
  LinkedInBuilder self() {
    return this;
  }

  @Override
  void startActivityForResult(Intent intent, int requestCode) {
    activity.startActivityForResult(intent, requestCode);
  }
}

File: linkedinsdk/src/main/java/com/AbdAllahAbdElFattah13/linkedinsdk/ui/linkedin_builder/LinkedInFromActivityBuilder.kt
package com.AbdAllahAbdElFattah13.linkedinsdk.ui.linkedin_builder

import android.app.Activity
import android.content.Intent

class LinkedInFromActivityBuilder private constructor(private val activity: Activity) : LinkedInBuilder(activity) {
    override fun self(): LinkedInBuilder {
        return this
    }

    override fun startActivityForResult(intent: Intent, requestCode: Int) {
        activity.startActivityForResult(intent, requestCode)
    }

    companion object {
        @JvmStatic
        fun getInstance(activity: Activity): LinkedInFromActivityBuilder {
            return LinkedInFromActivityBuilder(activity)
        }
    }

}